RiscVPkg/Include/Library/OpenSbiPeim.h

```diff
@@ -0,0 +1,340 @@
+/** @file
+  Defines the PEI_RISCV_OPENSBI_LEGACY_PPI to let PEIMs call OpenSBI
+
+Copyright (c) 2020, Hewlett Packard Development LP. All rights reserved.<BR>
+
+SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#ifndef _PEI_RISCV_OPENSBI_PPI_H_
+#define _PEI_RISCV_OPENSBI_PPI_H_
+
+#include <sbi/sbi_scratch.h>
+#include <sbi/sbi_platform.h>
+
+///
+/// Global ID for the PEI_RISCV_OPENSBI_BASE_PPI_GUID.
+///
+#define PEI_RISCV_OPENSBI_BASE_PPI_GUID  \
+  { \
+    0x9af5b3c3, 0x1e8c, 0x4219, { 0xa4, 0xfb, 0x3f, 0xb8, 0xb2, 0xb8, 0xfe, 0xa1 } \
+  }
+
+///
+/// Forward declaration for PEI_RISCV_OPENSBI_BASE_PPI.
+///
+typedef struct _PEI_RISCV_OPENSBI_BASE_PPI PEI_RISCV_OPENSBI_BASE_PPI;
+
+///
+/// SpecVersion:
+/// The minor number of the SBI specification is encoded in the low 24 bits,
+/// with the major number encoded in the next 7 bits.  Bit 32 must be 0 and is
+/// reserved for future expansion.
+///
+typedef
+EFI_STATUS
+(EFIAPI *PEI_RISCV_OPENSBI_GET_SPEC_VERSION)(
+  IN  EFI_PEI_SERVICES           **PeiServices,
+  IN  PEI_RISCV_OPENSBI_BASE_PPI  *This,
+  OUT UINTN                       *SpecVersion
+  );
+
+typedef
+EFI_STATUS
+(EFIAPI *PEI_RISCV_OPENSBI_GET_IMPL_ID)(
+  IN  EFI_PEI_SERVICES          **PeiServices,
+  IN  PEI_RISCV_OPENSBI_BASE_PPI *This,
+  OUT long                       *ImplId
+  );
+
+typedef
+EFI_STATUS
+(EFIAPI *PEI_RISCV_OPENSBI_GET_IMPLVERSION)(
+  IN  EFI_PEI_SERVICES           **PeiServices,
+  IN  PEI_RISCV_OPENSBI_BASE_PPI  *This,
+  OUT long                        *ImplVersion
+  );
+
+// TODO: Should we use an out parameter for Available or EFI_STATUS?
```

Comment:
cc @changab

Reply:
EFI_STATUS returned from this function is sufficient for the availability of SBI implementation version.

Reply:
Alright